tests/DCPS/Compiler/idl_test1_lib/FooDef.idl

```diff
@@ -53,11 +53,18 @@ module Xyz {
 
   typedef sequence<ArrayOfShorts> ArrayOfShortsSeq;
 
-#pragma DCPS_DATA_TYPE "Xyz::AStruct"
-#pragma DCPS_DATA_KEY "Xyz::AStruct v1"
   // +12
   typedef sequence<short> ShortSeq;
+#ifdef USE_TOPIC_ANNOTATIONS
+  @topic
+#else
+#  pragma DCPS_DATA_TYPE "Xyz::AStruct"
+#  pragma DCPS_DATA_KEY "Xyz::AStruct v1"
+#endif
   struct AStruct {
+#ifdef USE_TOPIC_ANNOTATIONS
```

Comment:
An alternative here would be to define a preprocessor macro that expands to either `@key` or to nothing.